src/components/transactions/TxDetails/TxData/ExecTransaction/index.tsx

```diff
@@ -0,0 +1,106 @@
+import useAsync from '@/hooks/useAsync'
+import NestedTransactionIcon from '@/public/images/transactions/nestedTx.svg'
+import ExternalLink from '@/components/common/ExternalLink'
+
+const safeInterface = Safe__factory.createInterface()
+
+const extractTransactionData = (data: string | undefined): SafeTransaction | undefined => {
+  const params = data ? safeInterface.decodeFunctionData('execTransaction', data) : undefined
+  if (!params || params.length !== 10) {
+    return
+  }
+
+  return {
+    addSignature: () => {},
+    encodedSignatures: () => params[9],
+    getSignature: () => undefined,
+    data: {
+      to: params[0],
+      value: params[1],
+      data: params[2],
+      operation: params[3] as number,
+      safeTxGas: params[4],
+      baseGas: params[5],
+      gasPrice: params[6],
+      gasToken: params[7],
+      refundReceiver: params[8],
+      nonce: 0,
+    },
+    signatures: new Map(),
+  }
+}
+
+export const ExecTransaction = ({ data }: { data?: TransactionData }) => {
+  const chain = useCurrentChain()
+
+  const childSafeTx = useMemo<SafeTransaction | undefined>(() => extractTransactionData(data?.hexData), [data?.hexData])
+
+  const [dataDecoded, error] = useAsync(async () => {
+    if (chain?.chainId && data?.to.value && childSafeTx) {
+      return await getConfirmationView(
+        chain.chainId,
+        data.to.value,
+        childSafeTx.data.data,
+        childSafeTx.data.to,
+        childSafeTx.data.value.toString(),
+      )
+    }
+  }, [chain?.chainId, data?.to.value, childSafeTx])
+
+  const decodedDataBlock = dataDecoded ? (
+    <DecodedTx tx={childSafeTx} showMethodCall decodedData={dataDecoded} showAdvancedDetails={false} />
+  ) : null
+
+  return (
+    <Stack spacing={2}>
+      {data?.dataDecoded && (
+        <>
+          <MethodCall contractAddress={data.to.value} method={data.dataDecoded.method} />
+          <MethodDetails data={data.dataDecoded} addressInfoIndex={data.addressInfoIndex} />
+        </>
+      )}
+
+      <Divider />
+
+      <Stack spacing={2}>
+        <Typography variant="h5" display="flex" alignItems="center" gap={1}>
+          <SvgIcon component={NestedTransactionIcon} inheritViewBox fontSize="small" /> Nested transaction:
+        </Typography>
+        {decodedDataBlock ? (
+          <>
+            {decodedDataBlock}
+
+            {chain && data && (
+              <Link
+                href={{
+                  pathname: AppRoutes.transactions.history,
+                  query: { safe: `${chain.shortName}:${data.to.value}` },
+                }}
+                passHref
+                legacyBehavior
+              >
+                <ExternalLink>Open Safe</ExternalLink>
+              </Link>
+            )}
+          </>
+        ) : error ? (
+          <ErrorMessage>Could not load details on executed transaction.</ErrorMessage>
+        ) : (
+          <Skeleton />
+        )}
+      </Stack>
+    </Stack>
+  )
+}
```

Comment:
There's a lot of similarity with the [`OnChainConfirmation` component](https://github.com/safe-global/safe-wallet-web/blob/c8b4f51089f77619413b8c01b339fb40df874d62/src/components/transactions/TxDetails/TxData/OnChainConfirmation/index.tsx). We could create a wrapper of sorts that you pass the transaction-to-display and reuse it for both.